src/control/system/errors.go

```diff
@@ -233,6 +234,37 @@ type ErrPoolNotFound struct {
 	byLabel *string
 }
 
+func (err *ErrPoolNotFound) MarshalJSON() ([]byte, error) {
```

Comment:
why is there a need for custom serialisation? rank and uuid have those methods already, maybe I am missing something else?

Reply:
The fields are unexported, intentionally. I decided that this was the least-worst solution to allow us to get the error on the dmg side.

Reply:
ah, why do they need to not be exported?

Reply:
> ah, why do they need to not be exported?

Because the intent is for the error types to be opaque outside of the package. That decision could be revisited at some point, but I didn't want to get into it for this PR. If we do change it so that the fields are exported, then we could remove the custom marshal/unmarshal methods on the type without any ripple effects.